sync15-adapter/src/driver.rs

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
It doesn't really need a relationship with FxA because the fxa stuff comes to us over android, and has to because of oauth reasons.

I think we probably do need a new crate for more robust sync functionality though, but it's not 100% clear to me the direction the dependencies flow. Does logins depend on it, or the other way around?